astropy/cosmology/funcs/tests/test_comparison.py

```diff
@@ -0,0 +1,242 @@
+                             [[None, format], [format, None]])
+        assert out.shape == (2, 2)
+        assert np.all(out == cosmo)
+
+    def test_parse_formats_vectorized(self, cosmo):
+        # vectorized on cosmos
+        out = _parse_formats(cosmo, cosmo, format=None)
+        assert len(out) == 2
+        assert np.all(out == cosmo)
+
+        # does NOT vectorize on formats
+        with pytest.raises(ValueError, match="operands could not be broadcast"):
+            _parse_formats(cosmo, format=[None, None])
+
+
+class Test_cosmology_equal(ComparisonFunctionTestBase):
+    """Test :func:`astropy.cosmology.comparison.cosmology_equal`"""
+
+    def test_cosmology_equal_simple(self, cosmo, pert_cosmo):
+        # equality
+        assert cosmology_equal(cosmo, cosmo) is True
+
+        # not equal to perturbed cosmology
+        assert cosmology_equal(cosmo, pert_cosmo) is False
+
+    def test_cosmology_equal_equivalent(self, cosmo, cosmo_eqvxflat,
+                                        pert_cosmo, pert_cosmo_eqvxflat):
+        # now need to check equivalent, but not equal, cosmologies.
+        assert cosmology_equal(cosmo, cosmo_eqvxflat, allow_equivalent=True) is True
+        assert cosmology_equal(cosmo, cosmo_eqvxflat, allow_equivalent=False) is False
+
+        assert cosmology_equal(pert_cosmo, pert_cosmo_eqvxflat, allow_equivalent=True) is True
+        assert cosmology_equal(pert_cosmo, pert_cosmo_eqvxflat, allow_equivalent=False) is False
+
+    def test_cosmology_equal_format_error(self, cosmo, converted):
+        # Not converting `converted`
+        with pytest.raises(TypeError):
+            cosmology_equal(cosmo, converted)
+
+        with pytest.raises(TypeError):
+            cosmology_equal(cosmo, converted, format=False)
+
+    def test_cosmology_equal_format_auto(self, cosmo, converted, xfail_cant_autoidentify):
+        # These tests only run if the format can autoidentify.
+        assert cosmology_equal(cosmo, converted, format=None) is True
+        assert cosmology_equal(cosmo, converted, format=True) is True
+
+    def test_cosmology_equal_format_specify(self, cosmo, format, converted, pert_converted):
+        # equality
+        assert cosmology_equal(cosmo, converted, format=[None, format]) is True
+        assert cosmology_equal(converted, cosmo, format=[format, None]) is True
+
+        # non-equality
+        assert cosmology_equal(cosmo, pert_converted, format=[None, format]) is False
+
+    def test_cosmology_equal_equivalent_format_specify(self, cosmo, format, converted, cosmo_eqvxflat):
+        # specifying the format
+        assert cosmology_equal(cosmo_eqvxflat, converted, format=[None, format], allow_equivalent=True) is True
+        assert cosmology_equal(converted, cosmo_eqvxflat, format=[format, None], allow_equivalent=True) is True
+
+
+class Test_cosmology_not_equal(ComparisonFunctionTestBase):
+    """Test :func:`astropy.cosmology.comparison._cosmology_not_equal`"""
```

Comment:
Does the hyperlink make sense if the function is private?
```suggestion
    """Test ``astropy.cosmology.comparison._cosmology_not_equal``"""
```

Reply:
Definitely less so. But the test suite isn't linked, so non-operable anyway. If this part of the test suite is ever made public it might break loudly and have to be fixed, versus remain unlinked quietly.